drivers/soundwire/bus_type.c

```diff
@@ -109,6 +109,8 @@ static int sdw_drv_probe(struct device *dev)
 
 	ret = drv->probe(slave, id);
 	if (ret) {
+		slave->ops = NULL;
```

Comment:
Should we set `slave->ops = NULL;` when `dev_pm_domain_attach` return error , too?

Reply:
yes indeed, thanks for spotting this @bardliao